core/src/test/java/nl/nn/adapterframework/stream/MessageTest.java

```diff
@@ -892,4 +909,66 @@ public void testNullMessageIsEmpty() {
 		Message message = null;
 		assertTrue(Message.isEmpty(message));
 	}
+
+	@Test
+	public void testMessageDefaultCharset() throws Exception {
+		String utf8Input = "Më-×m👌‰Œœ‡TzdDEyMt120=";
+		ByteArrayInputStream source = new ByteArrayInputStream(utf8Input.getBytes("utf-8"));
+		Message binaryMessage = new Message(source); //non-repeatable stream, no provided charset
+
+		assertEquals(utf8Input, binaryMessage.asString()); //Default must be used
+
+		Message characterMessage = new Message(utf8Input);
+
+		assertEquals(utf8Input, characterMessage.asString("ISO-8859-1")); //This should not be used as there is no binary conversion
+	}
+
+	@Test
+	public void testMessageDetectCharset() throws Exception {
+		String utf8Input = "Më-×m👌‰Œœ‡TzdDEyMt120=";
+		ByteArrayInputStream source = new ByteArrayInputStream(utf8Input.getBytes("utf-8"));
+		Message message = new Message(source, "auto"); //Set the MessageContext charset
+
+		String stringResult = message.asString("ISO-8859-ik-besta-niet"); //use MessageContext charset
+		assertEquals(utf8Input, stringResult);
+	}
+
+	@Test
+	public void testMessageDetectCharsetISO8859() throws Exception {
+		URL isoInputFile = TestFileUtils.getTestFileURL("/Util/MessageUtils/iso-8859-1.txt");
+		assertNotNull("unable to find isoInputFile", isoInputFile);
+
+		Message message = new UrlMessage(isoInputFile); //repeatable stream, detect charset
+
+		String stringResult = message.asString("auto"); //detect when reading
+		assertEquals(Misc.streamToString(isoInputFile.openStream(), "ISO-8859-1"), stringResult);
+	}
+
+	@Test
+	public void testCharsetDeterminationAndFallbackToDefault() throws Exception {
+		Message messageNullCharset = new Message((String) null) { //NullMessage, charset cannot be determined
+			@Override
```

Comment:
Is eigenlijk raar dat je hier een 'String' messag gebruikt. Die zou geen charset moeten hebben, en een empty message ook niet. Je test nu wel iets, maar computeDecodingCharset zou hier ook null op mogen leveren.

Reply:
Het is raar dat een String message een charset veld heeft. 
Hier wordt in puur de 'flow' van charset bepalen getest, en daar moet de 'compute' null teruggeven.